pkg/queue/manager.go

```diff
@@ -300,13 +300,14 @@ func (m *Manager) RequeueWorkload(ctx context.Context, info *workload.Info, imme
 		return false
 	}
 
-	q.AddIfNotPresent(info)
+	newInfo := workload.NewInfo(&w)
+	q.AddIfNotPresent(newInfo)
 	cq := m.clusterQueues[q.ClusterQueue]
 	if cq == nil {
 		return false
 	}
 
-	added := cq.RequeueIfNotPresent(info, immediate)
+	added := cq.RequeueIfNotPresent(newInfo, immediate)
```

Comment:
Should we change this for just `Requeue` instead of `RequeueIfNotPresent`?

It might not be necessary, as one of the routines trying to requeue will have the chance to push the newest version.

Reply:
sg.